Packs/ProofpointEmailSecurity/Integrations/ProofpointOnDemandEmailSecurityEventCollector/ProofpointEmailSecurityEventCollector.py

```diff
@@ -0,0 +1,96 @@
+from contextlib import contextmanager
+from enum import Enum
+from CommonServerPython import *  # noqa: F401
+from websockets.sync.client import connect
+from websockets.sync.connection import Connection
+from dateutil import tz
+
+VENDOR = "proofpoint"
+PRODUCT = "email_security"
+
+URL = "wss://{host}/v1/stream?cid={cluster_id}&type={type}&sinceTime={time}"
+
+
+EVENTS_TO_FETCH = 30
+TIMEOUT = 20
+
+
+class EventType(str, Enum):
+    MESSAGE = "message"
+    MAILLOG = "maillog"
+
+
+@contextmanager
+def websocket_connections(host: str, cluster_id: str, api_key: str):
+    now = datetime.utcnow().isoformat()
+    demisto.info("Starting websocket connection")
+    extra_headers = {"Authorization": f"Bearer {api_key}"}
+    with connect(
+        URL.format(host=host, cluster_id=cluster_id, type=EventType.MESSAGE.value, time=now),
+        additional_headers=extra_headers,
+    ) as message_connection, connect(
+        URL.format(host=host, cluster_id=cluster_id, type=EventType.MAILLOG.value, time=now),
+        additional_headers=extra_headers,
+    ) as maillog_connection:
+        yield message_connection, maillog_connection
+
+
+def fetch_events(event_type: EventType, connection: Connection, events_to_fetch: int, timeout: int) -> list[dict]:
+    events: list[dict] = []
+    while len(events) < events_to_fetch:
+        try:
+            event = json.loads(connection.recv(timeout=timeout))
+            event_ts = event.get("ts")
+            if not event_ts:
+                raise DemistoException(f"Event does not contain a timestamp: {event}")
+            date = dateparser.parse(event_ts)
+            if not date:
+                raise DemistoException(f"Failed to parse date: {event_ts}")
+            # the `ts` parameter is not always in UTC, so we need to convert it
+            event["_time"] = date.astimezone(tz.tzutc()).isoformat()
+            events.append(event)
+            demisto.debug(f"Received event. length of events from {event_type} is: {len(events)}")
+        except TimeoutError:
+            demisto.debug(f"Reached the end of time windows to collect events. Collected {len(events)} events from {event_type}")
```

Comment:
It will always be the same length, maybe change to how many inserted with dummy time/ skipped.